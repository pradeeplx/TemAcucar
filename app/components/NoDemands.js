import React from 'react'
import { fontFactor } from "../helpers"
import Colors from "../Colors"
import Sentence from "./Sentence"
import Tip from "./Tip"

export default NoDemands = ({ text }) => (
  <Tip style={{borderTopWidth: 0}}>
    <Sentence style={{
      alignSelf: 'stretch',
      marginHorizontal: 10,
      textAlign: 'center',
      fontSize: 10 * fontFactor(),
    }}>
      { `${text || 'Você ainda não recebeu pedidos dos seus vizinhos'}. Que tal agitar sua vizinhança pedindo algo que você precisa?` }
    </Sentence>
  </Tip>
)
