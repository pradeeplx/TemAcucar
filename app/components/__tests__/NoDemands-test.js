import 'react-native'
import React from 'react'
import NoDemands from '../NoDemands'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <NoDemands text="Foo Bar" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders default text', () => {
  const tree = renderer.create(
    <NoDemands />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
