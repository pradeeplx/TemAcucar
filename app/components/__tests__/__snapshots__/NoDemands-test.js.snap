exports[`test renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#FAF6F1",
        "borderTopWidth": 12,
        "padding": 15,
      },
      Object {
        "borderTopWidth": 0,
      },
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Avenir",
          "fontSize": 18,
        },
        Object {
          "alignSelf": "stretch",
          "fontSize": 15,
          "marginHorizontal": 10,
          "textAlign": "center",
        },
      ]
    }>
    Foo Bar. Que tal agitar sua vizinhança pedindo algo que você precisa?
  </Text>
</View>
`;

exports[`test renders default text 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#FAF6F1",
        "borderTopWidth": 12,
        "padding": 15,
      },
      Object {
        "borderTopWidth": 0,
      },
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Avenir",
          "fontSize": 18,
        },
        Object {
          "alignSelf": "stretch",
          "fontSize": 15,
          "marginHorizontal": 10,
          "textAlign": "center",
        },
      ]
    }>
    Você ainda não recebeu pedidos dos seus vizinhos. Que tal agitar sua vizinhança pedindo algo que você precisa?
  </Text>
</View>
`;
